### ISSUE TITLE
About training and prediction

### ISSUE THREAD

**Issue by VishwaasHegde** [marl/crepe#66]
Hello,
First of all thanks for the amazing paper and the repo !!
I have a basic doubt, the [RWC Dataset](https://staff.aist.go.jp/m.goto/RWC-MDB/#intro) says that annotated data is at semi-tone intervals, that is 50 cents.
How is CREPE able to predict with 10 or 20 cent intervals?

**Comment by VishwaasHegde** [comment]
Thanks, also since you are taking just one pitch output for a frame, why are you taking 'sigmoid' activation in `Dense(360, activation='sigmoid', name="classifier")` for the output, would 'softmax' be a better option? I believe sigmoid is usually used for multi label classification, whereas this is multi-class classification

**Comment by jongwook** [comment]
It's one of the tricks used for the approach, which is not quite orthodox for classification tasks in ML - it also uses binary cross entropy with soft labels, whereas the labels are usually one-hot in classification models.

We found that this combination (binary cross entropy with soft label) worked more robustly on pitch estimation, combined with the decoding heuristics taking the weighted average of activations near argmax.

**Comment by VishwaasHegde** [comment]
Thanks for the info. May I ask how you were able to obtain soft labels? Was it labelled that way in the data itself? I have a similar dataset that has hard pitch frequency labels. The only way I can think of taking soft labels is by having a Gaussian around each pitch frequency and with a standard deviation of 5-10 cents

**Comment by jongwook** [comment]
The labels I had contained Hz values (that doesn't necessarily align with semitone intervals), from which I calculated the soft labels using a Gaussian-shaped curve with a standard deviation of 25 cents. You can find an example code in the comments of [this issue](https://github.com/marl/crepe/issues/2#issuecomment-373288691).

**Comment by VishwaasHegde** [comment]
Hello, sorry for troubling you again. I have built a tonic and mode prediction model on top of CREPE, and I am looking to evaluate my tonic model. I understand that you have used `mir_eval` and `mir_eval.melody.raw_pitch_accuracy`. But I'm not sure how to use the api. I have tonic output and the target values in the cents and frequency format. I've searched online and have not found any resource as to how to use it correctly. It would be great if you could provide me some info on this. Thanks
